### PR TITLE
Feature - Admin endorse PPL

### PR DESCRIPTION
**Pull request by joefitter**
* Added new statuses with-admin and admin-endorsed
* admin-endorsed auto-forwards to with-inspectorate
* if applicant is admin, project grant/update go to inspectorate, or to with-admin if not

### CODE REVIEW

File: lib/query-builders/admin.js [hunk 1]
```diff
@@ -23,7 +24,8 @@ module.exports = {
       .where(buildEstablishmentQuery(profile))
       .whereIn('status', [
         returnedToApplicant.id,
-        recalledByApplicant.id
+        recalledByApplicant.id,
+        withAdmin.id
```
Comment (lennym) [inline lib/query-builders/admin.js]: I would probably not put this in here and would instead add a separate query builder for endorsement that applies to anyone who meets the `canEndorse` criteria.

`./lib/query-builders/endorse.js`

File: lib/flow/index.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ const {
   updated,
   resubmitted,
   returnedToApplicant,
+  withAdmin,
+  adminEndorsed,
```
[the remark this answers is not in the document]
Comment (joefitter) [reply lib/flow/index.js]: 👍

File: lib/hooks/create/project.js [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
   const messager = Messager(settings);
 
+  function checkAdmin(profileId, establishmentId) {
+    return Promise.resolve()
+      .then(() => Profile.query().eager('establishments').findById(profileId))
+      .then(profile => {
+        const role = profile.establishments.find(e => e.id === establishmentId).role;
+        return role === 'admin';
+      });
+  }
```
[the remark this answers is not in the document]
Comment (joefitter) [reply lib/hooks/create/project.js]: 👍

File: lib/query-builders/admin.js (hunk 1 above)
Comment (joefitter) [reply lib/query-builders/admin.js]: 🌮

File: lib/util.js [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+      return false;
   }
+}
+
+function getEstablishmentsWhereUserIsRole(role, profile) {
+  return profile.roles.map(r => (r.type === role) ? r.establishmentId : null);
+}
+
+function getEstablishmentsWhereUserCanEndorsePPL(profile) {
+  return profile.establishments.map(e => (e.role === 'admin') ? e.id : null);
```
Comment (lennym) [inline lib/util.js]:
```suggestion
  return profile.establishments.filter(e => e.role === 'admin').map(e => e.id);
```

File: lib/util.js [hunk 5]
```diff
@@ -1,5 +1,30 @@
...
+      return !!profile.roles.find(role => role.establishmentId === establishmentId && role.type === 'ntco');
+    case 'project':
+      return profile.establishments.find(e => e.id === establishmentId).role === 'admin';
+    default:
+      return false;
   }
+}
+
+function getEstablishmentsWhereUserIsRole(role, profile) {
+  return profile.roles.map(r => (r.type === role) ? r.establishmentId : null);
```
Comment (lennym) [inline lib/util.js]:
```suggestion
  return profile.roles.filter(r => r.type === role0.map(r => r.establishmentId);
```

**Pull request merged**